boot/boot.go

```diff
@@ -146,6 +146,13 @@ type bootState interface {
 	// the try snap (only the latter might not be set), and
 	// whether the snap is in "trying" state.
 	revisions() (snap, try_snap *NameAndRevision, trying bool, err error)
+
+	// markSuccessful lazily implements marking the boot
+	// successful for the boot snap of the type. actually
+	// committing the update is done via the threaded
+	// bootStateUpdate commit, that way different bootState
+	// markSuccessful can be chained.
```

Comment:
I want to be part of the bikesheed as well :) I like the "fold" but I wonder if it's a bit of a big word for something that is relatively straightforward. Maybe the the emphasis should be more on that `markSuccessful` collects and `commit` commits as atomic as possible what was collected in the bootState? But not a blocker, I think the folding comment is fine and so is just leaving threaded out (I agree that we should try to avoid it because of the os/goroutine connotation).